src/main/java/io/confluent/idesidecar/restapi/models/graph/ConfluentRestClient.java

```diff
@@ -220,21 +220,20 @@ protected record LastRequest(String url, String json) {
    * If the response is paginated, this stream will continue to get additional pages as
    * needed until there are no more results or the supplied limits are reached.
    *
-   * @param connectionId   the ID of the connection to use
+   * @param headers        the headers to use for the request
    * @param firstUrl       the URL of the first page of Confluent API resources
    * @param limits         the page limits; may be null if there are no limits
    * @param responseParser the parser for the items on the page
    * @param <T>            the type of items being returned
    * @return the stream of items
    */
   protected <T> Multi<T> listItems(
-      String connectionId,
+      MultiMap headers,
```

Comment:
The places where this method is used knows whether it's using the REST client for Kafka or SR. Since each type of cluster need different credentials and therefore authN-related headers, it's easier for the caller to compute the appropriate headers and simply pass them into this method.